dashboard/app/controllers/reports_controller.rb

```diff
@@ -56,6 +56,20 @@ def all_usage
     render 'usage', formats: [:html]
   end
 
+  def search_for_teachers
+    authorize! :read, :reports
+    filter = params[:filter]
+    fields = ['name', 'email']
+    teachers = User.find_by_sql(<<SQL)
+SELECT u.name, u.email
+FROM users AS u
+WHERE u.user_type = 'teacher'
+  AND u.id NOT IN (SELECT teacher_id FROM workshop_attendance)
+  AND u.full_address LIKE '%#{filter}%'
```

Comment:
I know this is an admin only page, so we aren't worried about malicious usage, but I think it's good practice to use the sql specific replacement instead of string interpolation so it can be sanitized: http://apidock.com/rails/ActiveRecord/Querying/find_by_sql